### ISSUE TITLE
Normalizing CIF output takes more time than running CIF

### ISSUE THREAD

**Issue by eunovm** [17451k/clade#62]
According to:
```
10:35:55 clade Info: Start CIF                                                 
10:35:55 clade Info: Parsing 1964 commands
10:42:28 clade Info: CIF finished with errors. Log: /home/novikov/work/klever-data/linux-3.14.79/clade/Info/err.log
10:42:28 clade Info: Normalizing CIF output
10:50:46 clade Info: Normalizing finished
```
normalizing CIF output took 8 minutes 18 seconds while running CIF took only 6 minutes 33 seconds.

**Comment by eunovm** [comment]
When the number of parsed commands grows the issue hurts much more:
```
12:17:48 clade Info: Start CIF                                                 
12:17:49 clade Info: Parsing 13846 commands
13:26:30 clade Info: CIF finished with errors. Log: /home/novikov/work/klever-data/linux-3.14.79/clade/Info/err.log
13:26:30 clade Info: Normalizing CIF output
14:50:59 clade Info: Normalizing finished
```

Note that normalizing CIF output took almost 10 times more than running CIF.

**Comment by 17451k** [comment]
I am well aware about it, but I don't know how to make in faster.

**Comment by eunovm** [comment]
We can calculate some hashes in CIF in advance so that it will be easier to match corresponding data. Perhaps we need to output query results into different files, e.g. at least for each source directory or even each individual source file.

But you need to ensure that the method will do help Clade (we need linear dependency rather than polynomial one) since it is not so easy to implement something in CIF.

**Comment by 17451k** [comment]
Thanks to recent changes now the normalisation process is up to 4 times faster, but CIF itself is slower by 30%. Overall speed increase is by 20%.

Additional optimisations will be performed inside CIF.

**Comment by eunovm** [comment]
By some reason commit 1651616 to CIF that just fixed a small bug slows down normalization very considerably. Can you investigate this strange behavior?

**Comment by eunovm** [comment]
Likely I identified the issue. After that commit that changed completely another thing, CIF started to print function signatures incorrectly. Instead of, say, valid:

> void apic_set_eoi_write(void (*)(u32, u32));

it outputs invalid:

> void apic_set_eoi_write(void(*)(u32, u32);

I will fix the issue in CIF.

**Comment by eunovm** [comment]
No, that was not the issue. It was already fixed in CIF and this didn't help. The real issue remains unknown. I did not find anyd changes that can slow down normalization that seems to be trivial.

**Comment by eunovm** [comment]
The only thing I can definitely say is that normalization takes 4 times more time when launching Clade from scratch. With removing "Info" or command-line option "-fc" for _clade-info_ Clade normalization operates fast enough. I have no ideas, why fixing 

Besides, the new approach for storing CIF output and normalization is broken on Windows. CIF doesn't create subdirectories within "Info/output" sometimes and some dumped data is not processed well by Clade. Perhaps this is related with #79, but I am not sure.

**Comment by eunovm** [comment]
The latter was fixed in bb3ffeb98b2a943c48cacb4276625e00be595c13. The former should be investigated further one day latter. It is not so important.
